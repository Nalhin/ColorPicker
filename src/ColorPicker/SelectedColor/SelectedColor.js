import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { colorArrayToRgba } from '../Utils';
import { SELECTED_COLOR_SIZE } from '../constants';

const Background = styled.div`
  width: ${SELECTED_COLOR_SIZE}px;
  height: ${SELECTED_COLOR_SIZE}px;
  border-radius: 50%;
`;

const SelectedColor = ({ color }) => {
  return <Background style={{ background: colorArrayToRgba(color) }} />;
};

SelectedColor.propTypes = {
  color: PropTypes.arrayOf(PropTypes.number).isRequired,
};

export default SelectedColor;
