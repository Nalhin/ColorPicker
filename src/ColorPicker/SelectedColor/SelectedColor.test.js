import React from 'react';
import { render } from '@testing-library/react';
import SelectedColor from './SelectedColor';

describe('Selected Color', () => {
  // toHaveStyles doesn't work with rgba values

  it('Should render correctly', () => {
    const color = [255, 240, 220, 1];
    const { container } = render(<SelectedColor color={color} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  // it('Should change background color', () => {
  //   const color = 'rgb(140, 52, 30)';
  //   const { container } = render(<SelectedColor color={color} />);

  //   expect(container).toHaveStyleRule('background', color);
  // });

  // it('Should render with default color', () => {
  //   const defaultColor = 'rgb(0, 0, 0)';
  //   const { container } = render(<SelectedColor />);
  //   expect(container).toHaveStyleRule(`background: ${defaultColor}`);
  // });
});
