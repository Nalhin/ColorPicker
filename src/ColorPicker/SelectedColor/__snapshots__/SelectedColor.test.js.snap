// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected Color Should render correctly 1`] = `
<div
  class="css-18vytbf-SelectedColor__Background e1svbpgc0"
  style="background: rgb(255, 240, 220);"
/>
`;
